### PR TITLE
enable/disable OpenShift depends on admin settings

### LINKED ISSUE #5328: Enable/disable OpenShift depends on admin settings
**User Story**
Enable/disable OpenShift in Create Cluster endpoint depends on admin settings.
Use enum:
```
ClusterTypeOptions {
  All = 0,
  Kubernetes = 1,
  OpenShift = 2,
}

```

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: enable/disable OpenShift, Kubermatic cluster creation depends on admin settings

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5328

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/handler/v1/cluster/cluster.go [hunk 1]
```diff
@@ -1043,17 +1045,23 @@ type CreateReq struct {
...
-		}
+	if !clusterTypes.Has(r.Body.Cluster.Type) {
+		return fmt.Errorf("invalid cluster type %s", r.Body.Cluster.Type)
 	}
-	return fmt.Errorf("invalid cluster type %s", r.Body.Cluster.Type)
+
+	if clusterType == kubermaticv1.ClusterTypeAll {
+		return nil
+	}
+	if clusterType != apiv1.ToInternalClusterType(r.Body.Cluster.Type) {
```
Comment (floreks) [inline api/pkg/handler/v1/cluster/cluster.go]: Merge 2 ifs into one `if clusterType != ClusterTypeAll && clusterType != internalClusterType`

File: api/pkg/api/v1/types.go [hunk 2]
```diff
@@ -640,6 +640,8 @@ const (
 	OpenShiftClusterType string = "openshift"
 	// KubernetesClusterType defines the Kubernetes cluster type
 	KubernetesClusterType string = "kubernetes"
+	// AllClusterTypes defines support for all cluster types
+	AllClusterTypes string = "all"
```
Comment (moadqassem) [inline api/pkg/api/v1/types.go]: This field here is not used. I am not sure if it was added to replace or interpret `kubermaticv1.ClusterTypeAll`, which you are using at line 1605(https://github.com/kubermatic/kubermatic/pull/5337/files?file-filters%5B%5D=.go#diff-7f8b07edb0cdb9a092e601081f5ddc8bR1605)

Comment (zreigz) [reply api/pkg/api/v1/types.go]: we have API and internal CRD for the cluster types. To make it comparable I had to add this new type for this https://github.com/kubermatic/kubermatic/pull/5337/files#diff-7f8b07edb0cdb9a092e601081f5ddc8bR1598

Comment (zreigz) [reply api/pkg/api/v1/types.go]: but you are right it's not used anywhere and we can remove it

**Comment by kdomanski** [comment]
:facepalm: 
/cherry-pick release/v2.14

**Pull request merged**